src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -104,11 +104,6 @@ public override string GetMessage(IFormatProvider? formatProvider = null)
                     return _original.GetMessage(formatProvider);
                 }
 
-                public override bool Equals(object? obj)
-                {
-                    return _original.Equals(obj);
```

Comment:
Was this implementation the bug at the root of #58954?

Reply:
it's half of it.  we'll still need to do the cast to CompilationAnalyzer in the override that still exists here.

Reply:
but importantly, there is just one place where we do the other downcasting.  teh primary downcast happens uniformly in one place and shouldn't be overridden.